File: Civitas_GUI/src/civitas/Casilla_Impuesto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author alejandro
 */
public class Casilla_Impuesto extends Casilla{
    private float importe;
    
    Casilla_Impuesto(float cantidad, String name){
        super(name);
        importe = cantidad;
    }
    
    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos){
        if (super.jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            todos.get(actual).pagaImpuesto(importe);
            todos.get(actual).setPuedeComprar(false);
        }
    }
    
    private void informe(int actual, ArrayList<Jugador> todos) {
        if (super.jugadorCorrecto(actual, todos)) {
            Diario.getInstance().ocurreEvento(todos.get(actual).getNombre() + " ha caido en el impuesto de " + this.toString());
        }
    }
    
    @Override
    public String toString(){
        String cadena = nombre + "\n Importe: " + importe;        
        return cadena;
    }
    
    public float get_importe(){
        return importe;
    }
}

File: Civitas_Ruby/lib/casilla_impuesto.rb
require_relative "titulo_propiedad.rb"
require_relative "sorpresa.rb"
require_relative "mazo_sorpresas"

module Civitas
  class Casilla_Impuesto < Casilla
    
    def initialize (cantidad, nombre)
      super(nombre)
      @importe = cantidad
    end
    
    def recibeJugador(actual, todos)
      if(jugadorCorrecto(actual, todos))
        informe(actual, todos)
        todos[actual].pagaImpuesto(@importe)
      end
    end
    
    def to_s()
      cadena = "#{@nombre} Precio: #{@importe}"
      return cadena
    end
    
    def informe(actual, todos)
      if(jugadorCorrecto(actual,todos))
        Diario.instance.ocurre_evento(todos[actual].nombre.to_s + " ha caido en el impuesto de " + to_s)

      end
    end
    
  end
end